shared/domain/src/main/java/com/ivy/domain/usecase/account/AccountBalanceUseCase.kt

```diff
@@ -33,11 +44,35 @@ class AccountBalanceUseCase @Inject constructor(
     suspend fun calculate(
         account: AccountId,
     ): Map<AssetCode, NonZeroDouble> {
-        TODO("Not implemented")
+        val accountStats = accountStatsUseCase.calculate(
+            account = account,
+            transactions = transactionRepository.findAll()
+        )
+
+        val balance = BalanceBuilder()
```

Comment:
To make it more Kotlin idiomatic we can:
```kotlin
return BalanceBuilder().run {
   processDeposits(...)
   processWithdtwqlas(...)
   build()
}
```

The code would be shorter and read nicer IMO